### PR TITLE
Added count_bits_flip.cpp

### PR DESCRIPTION
**Pull request by yashrajyash**
#### Description of Change

Program to count number of bits needed to be flipped to convert ‘A’ to ‘B’.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:  Added an algorithm which returns the count of number of bits to be flipped of an a number A to form the number B

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1862"></a>

### CODE REVIEW

**Review by Qannaf — APPROVED** [review]

**Comment by yashrajyash** [comment]
@Panquesito7 Please review my PR

File: bit_manipulation/count_bits_flip.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+namespace count_bits_flip {
+/**
+ * @brief The main function implements count of bits flip required
+ * @param A is the given number whose bits will be flipped to get number B
+ * @param B is the given target number
+ * @returns total number of bits needed to be flipped to convert A to B
+ */
+std::uint64_t countBitsFlip(
+    std ::int64_t A,
+    std ::int64_t B) {  // int64_t is preferred over int so that
```
Comment (Panquesito7) [inline bit_manipulation/count_bits_flip.cpp:44]:
```suggestion
    std::int64_t A,
    std::int64_t B) {  // int64_t is preferred over int so that
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 😄👍
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/yashrajyash/C-Plus-Plus/actions

**Comment by yashrajyash** [comment]
> Great work! 😄👍 Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/yashrajyash/C-Plus-Plus/actions

Sir, I've enabled GitHub Actions

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍

**Review by mishraabhinn — APPROVED** [review]

**Pull request merged**